internal/transactions/transactions.go

```diff
@@ -57,6 +58,77 @@ type TransactionResult struct {
 	exclude []string
 }
 
+type jsonValue interface{}
+type jsonValueObject struct {
+	Type  string    `json:"type"`
+	Value jsonValue `json:"value"`
+}
+type jsonCompositeField struct {
+	Name  string    `json:"name"`
+	Value jsonValue `json:"value"`
+}
+type jsonCompositeValue struct {
+	ID     string               `json:"id"`
+	Fields []jsonCompositeField `json:"fields"`
+}
+type jsonLegacyTypeValue struct {
+	StaticType string `json:"staticType"`
+}
+
+func prepareEvent(v cadence.Event) jsonValue {
+	return prepareComposite("Event", v.EventType.ID(), v.EventType.Fields, v.Fields)
+}
+
+func prepareComposite(kind, id string, fieldTypes []cadence.Field, fields []cadence.Value) jsonValue {
```

Comment:
could we just add a comment explaining why this workaround is necessary and potentially link it to an issue?

Reply:
Sure I will add today.